userservice/src/main/java/com/gaurav/userservice/UserserviceApplication.java

```diff
@@ -1,13 +1,36 @@
 package com.gaurav.userservice;
 
+import com.gaurav.userservice.entity.Role;
+import com.gaurav.userservice.entity.User;
+import com.gaurav.userservice.repository.UserRepository;
+import jakarta.annotation.PostConstruct;
+import jakarta.transaction.Transactional;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.security.crypto.password.PasswordEncoder;
 
 @SpringBootApplication
 public class UserserviceApplication {
 
+    private UserRepository userRepository;
+    private PasswordEncoder passwordEncoder;
+
+    public UserserviceApplication(UserRepository userRepository, PasswordEncoder passwordEncoder) {
+        this.userRepository = userRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
+
     public static void main(String[] args) {
         SpringApplication.run(UserserviceApplication.class, args);
     }
 
+    @PostConstruct
```

Comment:
Consider adding an environment-based conditional check to prevent this from running in production (e.g., if (env.equals("dev"))).

Alternatively, use database migrations (like Flyway or Liquibase) for managing initial data loads in a more controlled manner.